service/worker/deployment/deployment_workflow.go

```diff
@@ -126,22 +135,39 @@ func (d *DeploymentWorkflowRunner) run() error {
 			}
 
 			// Add the task queue to the local state
-			newTaskQueueWorkerInfo := &deployspb.DeploymentLocalState_TaskQueueFamilyInfo_TaskQueueInfo{
-				TaskQueueType:   updateInput.TaskQueueType,
-				FirstPollerTime: updateInput.FirstPollerTime,
+			newTaskQueueWorkerInfo := &deploymentspb.DeploymentLocalState_TaskQueueFamilyInfo_TaskQueueInfo{
+				TaskQueueType: updateInput.TaskQueueType,
+				Data: &deploymentspb.Data{
+					FirstPollerTime: updateInput.FirstPollerTime,
+				},
 			}
 			d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName].TaskQueues[int32(updateInput.TaskQueueType)] = newTaskQueueWorkerInfo
 
-			// Call activity which starts a "DeploymentSeries" workflow
-			return d.invokeDeploymentSeriesActivity(ctx, d.DeploymentLocalState.WorkerDeployment.SeriesName)
```

Comment:
originally it was because I wanted to split the execute and the wait on the future since I did the activities in parallel. it's more flexible to just work with the contexts and futures directly, so you can do things like starting activities in parallel, creating an intermediate cancel context shared by only certain activities, etc.

I changed them back to serial though, so I'm not using that flexibility anymore.

in general I think lots of small functions makes it harder to read/navigate code, since you have to keep jumping around to understand what's happening. I'm not opposed to factoring out some functions later to clean things up, but while it's in flux, I think this is easier to work with.

Reply:
not going to block you/spur more discussion here since there are more important problems to consider right now - but I do have some thoughts and we can discuss this later